cli/cli_app.go

```diff
@@ -338,6 +401,8 @@ func runTerragruntWithConfig(terragruntOptions *options.TerragruntOptions, terra
 		}
 	}
 
+	beforeHookErrors := processHooks(terragruntConfig.Terraform.GetBeforeHooks(), terragruntOptions)
```

Comment:
This is a fairly big behavioral change. Some use cases will want hooks to run before `init` and some after `init` but before the actual Terraform command. Therefore, the suggestion is to add a new `before_init_hook` type. This sort of change should be broken out into a separate PR.

Reply:
Sound reasonable and I guess that's why most systems do have a PreInit step or the like :-D